components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java

```diff
@@ -169,8 +169,8 @@ public boolean doPostSetUserClaimValueWithID(String s, UserStoreManager userStor
     public boolean doPreSetUserClaimValueWithID(String userID, String claimURI, String claimValue, String profileName,
                                                 UserStoreManager userStoreManager) throws UserStoreException {
 
-        // Validate whether claim update request is for a provisioned user.
-        if (FrameworkUtils.isJITProvisionEnhancedFeatureEnabled()) {
+        if (FrameworkUtils.isJITProvisionEnhancedFeatureEnabled() && StringUtils.isNotBlank(claimURI) &&
+                !claimURI.contains(UserCoreConstants.ClaimTypeURIs.IDENTITY_CLAIM_URI)) {
```

Comment:
Can't we use `isIdentityClaimUpdate` here as well?

Reply:
Fixed https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/376/commits/885367b448d53ffdb5414c784b8fccf532677ebb